mypy/meet.py

```diff
@@ -138,9 +140,8 @@ class C(A, B): ...
 
     # We must check for TupleTypes before Instances, since Tuple[A, ...]
     # is an Instance
-    tup_overlap = is_overlapping_tuples(t, s, use_promotions)
-    if tup_overlap is not None:
-        return tup_overlap
+    if is_overlapping_tuples(t, s, use_promotions):
+        return True
```

Comment:
I think this change is now unnecessary (and the corresponding change below) after you added the check at the top of this function.